react/Step.tsx

```diff
@@ -0,0 +1,86 @@
+import classNames from 'classnames'
+import React, { useEffect, useLayoutEffect, useRef } from 'react'
+
+import { useStepContext } from './StepGroup'
+
+const useStepIndicator = (elementRef: React.RefObject<HTMLLIElement>) => {
+  const indexRef = useRef<number>(-1)
+  const { assigning, items, updateParent, ...context } = useStepContext()
+
+  useLayoutEffect(() => {
+    if (!assigning.current) {
+      return
+    }
+
+    indexRef.current = items.current.push(elementRef.current!) - 1
+    updateParent()
+  })
+
+  return {
+    // first render its wrong, after a forceUpdate in parent useLayoutEffect it's
+    // right, and its all synchronous so we don't get any flashing
```

Comment:
I would rephrase this as something like
`index is wrong on the first render, but the parent forces a synchronous second render that fixes its value without flashing the component`

Reply:
nice